### ISSUE TITLE
[Bug] NPM install error

### ISSUE THREAD

**Issue by dherault** [xenova/transformers.js#85]
**Describe the bug**
*A clear and concise description of what the bug is.*

I get the following error while installing the package:
```
 npm i -S @xenova/transformers
npm ERR! code 1
npm ERR! path /Users/sven/dev/AMJS-2/transformers-demo/node_modules/canvas
npm ERR! command failed
npm ERR! command sh -c node-pre-gyp install --fallback-to-build --update-binary
npm ERR! Failed to execute '/Users/sven/.nvm/versions/node/v18.15.0/bin/node /Users/sven/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --update-binary --module=/Users/sven/dev/AMJS-2/transformers-demo/node_modules/canvas/build/Release/canvas.node --module_name=canvas --module_path=/Users/sven/dev/AMJS-2/transformers-demo/node_modules/canvas/build/Release --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v108' (1)
npm ERR! node-pre-gyp info it worked if it ends with ok
npm ERR! node-pre-gyp info using node-pre-gyp@1.0.10
npm ERR! node-pre-gyp info using node@18.15.0 | darwin | arm64
npm ERR! node-pre-gyp http GET https://github.com/Automattic/node-canvas/releases/download/v2.11.2/canvas-v2.11.2-node-v108-darwin-unknown-arm64.tar.gz
npm ERR! node-pre-gyp ERR! install response status 404 Not Found on https://github.com/Automattic/node-canvas/releases/download/v2.11.2/canvas-v2.11.2-node-v108-darwin-unknown-arm64.tar.gz 
npm ERR! node-pre-gyp WARN Pre-built binaries not installable for canvas@2.11.2 and node@18.15.0 (node-v108 ABI, unknown) (falling back to source compile with node-gyp) 
npm ERR! node-pre-gyp WARN Hit error response status 404 Not Found on https://github.com/Automattic/node-canvas/releases/download/v2.11.2/canvas-v2.11.2-node-v108-darwin-unknown-arm64.tar.gz 
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.3.1
npm ERR! gyp info using node@18.15.0 | darwin | arm64
npm ERR! gyp info ok 
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.3.1
npm ERR! gyp info using node@18.15.0 | darwin | arm64
npm ERR! gyp info find Python using Python version 3.9.6 found at "/Library/Developer/CommandLineTools/usr/bin/python3"
npm ERR! gyp http GET https://nodejs.org/download/release/v18.15.0/node-v18.15.0-headers.tar.gz
npm ERR! gyp http 200 https://nodejs.org/download/release/v18.15.0/node-v18.15.0-headers.tar.gz
npm ERR! gyp http GET https://nodejs.org/download/release/v18.15.0/SHASUMS256.txt
npm ERR! gyp http 200 https://nodejs.org/download/release/v18.15.0/SHASUMS256.txt
npm ERR! gyp info spawn /Library/Developer/CommandLineTools/usr/bin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/Users/sven/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/sven/dev/AMJS-2/transformers-demo/node_modules/canvas/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/sven/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/sven/Library/Caches/node-gyp/18.15.0/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/Users/sven/Library/Caches/node-gyp/18.15.0',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/Users/sven/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/Users/sven/Library/Caches/node-gyp/18.15.0/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/Users/sven/dev/AMJS-2/transformers-demo/node_modules/canvas',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! Package pixman-1 was not found in the pkg-config search path.
npm ERR! Perhaps you should add the directory containing `pixman-1.pc'
npm ERR! to the PKG_CONFIG_PATH environment variable
npm ERR! No package 'pixman-1' found
npm ERR! gyp: Call to 'pkg-config pixman-1 --libs' returned exit status 1 while in binding.gyp. while trying to load binding.gyp
npm ERR! gyp ERR! configure error 
npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
npm ERR! gyp ERR! stack     at ChildProcess.onCpExit (/Users/sven/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:325:16)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
npm ERR! gyp ERR! System Darwin 22.2.0
npm ERR! gyp ERR! command "/Users/sven/.nvm/versions/node/v18.15.0/bin/node" "/Users/sven/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--update-binary" "--module=/Users/sven/dev/AMJS-2/transformers-demo/node_modules/canvas/build/Release/canvas.node" "--module_name=canvas" "--module_path=/Users/sven/dev/AMJS-2/transformers-demo/node_modules/canvas/build/Release" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v108"
npm ERR! gyp ERR! cwd /Users/sven/dev/AMJS-2/transformers-demo/node_modules/canvas
npm ERR! gyp ERR! node -v v18.15.0
npm ERR! gyp ERR! node-gyp -v v9.3.1
npm ERR! gyp ERR! not ok 
npm ERR! node-pre-gyp ERR! build error 
npm ERR! node-pre-gyp ERR! stack Error: Failed to execute '/Users/sven/.nvm/versions/node/v18.15.0/bin/node /Users/sven/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --update-binary --module=/Users/sven/dev/AMJS-2/transformers-demo/node_modules/canvas/build/Release/canvas.node --module_name=canvas --module_path=/Users/sven/dev/AMJS-2/transformers-demo/node_modules/canvas/build/Release --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v108' (1)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/sven/dev/AMJS-2/transformers-demo/node_modules/@mapbox/node-pre-gyp/lib/util/compile.js:89:23)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
npm ERR! node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1091:16)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:302:5)
npm ERR! node-pre-gyp ERR! System Darwin 22.2.0
npm ERR! node-pre-gyp ERR! command "/Users/sven/.nvm/versions/node/v18.15.0/bin/node" "/Users/sven/dev/AMJS-2/transformers-demo/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--update-binary"
npm ERR! node-pre-gyp ERR! cwd /Users/sven/dev/AMJS-2/transformers-demo/node_modules/canvas
npm ERR! node-pre-gyp ERR! node -v v18.15.0
npm ERR! node-pre-gyp ERR! node-pre-gyp -v v1.0.10
npm ERR! node-pre-gyp ERR! not ok

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/sven/.npm/_logs/2023-04-13T14_37_05_105Z-debug-0.log
```

**How to reproduce**
*Steps or a minimal working example to reproduce the behavior*

`npm i -S @xenova/transformers` on NodeJS LTS 18.x

**Expected behavior**
*A clear and concise description of what you expected to happen.*

It installs correctly

**Logs/screenshots**
*If applicable, add logs/screenshots to help explain your problem.*

**Environment**
- Transformers.js version: latests
- Browser (if applicable):
- Operating system (if applicable): MacOS
- Other:

**Additional context**
*Add any other context about the problem here.*

**Comment by dherault** [comment]
Workaround: Install the last working version:
`npm i -S @xenova/transformers@1.4.1`

**Comment by xenova** [comment]
Are you able to install node canvas (https://www.npmjs.com/package/canvas) with `npm i canvas`? We switched from Jimp to canvas for loading images because Jimp is extremely bloated and slow.

**Comment by xenova** [comment]
According to the README on their github repo, you may need to install some extra dependencies:
https://github.com/Automattic/node-canvas#compiling

I will try to simplify this process so other's don't need to worry about this.

**Comment by xenova** [comment]
This might be relevant: https://github.com/Automattic/node-canvas/issues/2188

**Comment by xenova** [comment]
Just an update on this. I've also encountered installation issues when trying to build an electron app, since they don't come with precompiled binaries for the node.js version which comes with electron.

So, I'm switching to [sharp.js](https://github.com/lovell/sharp). I should have done this originally, since it's the [fastest](https://github.com/ivanoff/images-manipulation-performance) and most popular of the various backend image manipulation libraries. The original reason was the ensure the node.js and web versions had the same outputs, but it turns out that canvas doesn't even support some types of interpolation (which some models need). I'll eventually look into non-canvas ways to do fast image resizing in the browser.

**Comment by xenova** [comment]
Just posting here as well: 
I published a new version ([v1.4.3](https://www.npmjs.com/package/@xenova/transformers/v/1.4.3)) which fixes this.
